### PR TITLE
Point to 'latest' docs for install-devel

### LINKED ISSUE #2360: Instructions on installing development version should point to 'latest' docs
We have a section on installing the development version of cocotb in the last release. That doesn't make sense since the build instructions can change. We should instead point to the 'latest' docs installation instructions when linking to the installation of the development version. For example, changes in the installation of the development package afforded by #1372 won't be seen in the stable docs.

### PR DESCRIPTION
**Pull request by cmarqu**
Fixes #2360

### CODE REVIEW

File: documentation/source/install_devel.rst [hunk 1]
```diff
@@ -4,7 +4,17 @@
...
 **********************************
 
-The development version of cocotb may have differerent prerequisites
+.. note::
+
+   If you want to follow the instructions on this page,
+   make sure you are reading the version from the development version itself,
+   which is https://docs.cocotb.org/en/latest/install_devel.html
+
+   Once you installed the development version,
```
Comment (ktbarrett) [inline documentation/source/install_devel.rst:13]:
```suggestion
   Once you install the development version,
```

File: documentation/source/install_devel.rst [hunk 2]
```diff
@@ -4,7 +4,17 @@
 Installing the Development Version
 **********************************
 
-The development version of cocotb may have differerent prerequisites
+.. note::
+
+   If you want to follow the instructions on this page,
+   make sure you are reading the version from the development version itself,
+   which is https://docs.cocotb.org/en/latest/install_devel.html
```
Comment (ktbarrett) [inline documentation/source/install_devel.rst:11]:
```suggestion
   make sure you are reading the `latest version <https://docs.cocotb.org/en/latest/install_devel.html>`_.
```

File: documentation/source/install_devel.rst [hunk 3]
```diff
@@ -4,7 +4,17 @@
...
+
+   Once you installed the development version,
+   you should keep reading the matching documentation,
+   i.e. stay at https://docs.cocotb.org/en/latest/
```
Comment (ktbarrett) [inline documentation/source/install_devel.rst:15]: Like this.
```suggestion
   you should keep reading the `matching documentation <https://docs.cocotb.org/en/latest/>`_.
```

**Review by ktbarrett — APPROVED** [review]
This conflicts with some changes in #1798, and seeing as this is the easier to rebase I'd like to merge that first.

**Comment by cmarqu** [comment]
> This conflicts with some changes in #1798, and seeing as this is the easier to rebase I'd like to merge that first.

Sure, I'll hold off.

**Review by ktbarrett — APPROVED** [review]

**Pull request merged**